guessing-game-v2/contracts/GuessingGame.sol

```diff
@@ -1,86 +1,133 @@
+    /// @param guessedNumber The number guessed by the user.
+    /// @param success True if the guess is correct, false otherwise.
+    /// @param feedback A message providing feedback on the guess.
+    event Guessed(address indexed user, uint256 guessedNumber, bool success, string feedback);
 
+    /// @notice Ensures that only the owner can call the modified function.
     modifier onlyOwner() {
-        require(msg.sender == owner, 'Only the owner can call this function');
+        require(msg.sender == owner, "Only the owner can call this function");
         _;
     }
 
+    /// @notice Ensures that only an admin can call the modified function.
+    modifier onlyAdmin() {
+        require(admins[msg.sender], "Only admins can call this function");
+        _;
+    }
+
+    /// @dev Sets the deploying address as the owner and initializes the game.
     constructor() {
         owner = msg.sender;
+        admins[owner] = true;
         resetCount = 0;
-        _resetSecretNumber(); // Initialize the secret number when the contract is deployed.
+        _resetSecretNumber();
     }
 
+    /// @notice Allows a user to make a guess at the secret number.
+    /// @dev Requires a fee; emits a Guessed event with the result.
+    /// @param _number The number being guessed by the user.
     function guess(uint256 _number) external payable {
-        require(_number > 0 && _number <= MAX_GUESS, 'Secret number should be between 1 and 1000');
-        require(msg.value == GUESS_FEE, 'You need to pay a fee of 0.443 ETH to make a guess');
+        require(_number > 0 && _number <= MAX_GUESS, "Number must be between 1 and 1000");
+        require(msg.value == GUESS_FEE, "Incorrect fee");
+        guessCount[msg.sender] += 1;
         totalGuesses += 1;
 
         if (lastResetTime > lastGuess[msg.sender]) {
-            lastDifference[msg.sender] = MAX_GUESS + 1; // Reset the last difference if the player is guessing after a secret no. reset.
+            lastDifference[msg.sender] = MAX_GUESS + 1;
         }
 
         string memory feedback;
 
         if (_number == secretNumber) {
             payable(msg.sender).transfer(address(this).balance);
-            feedback = "correct! You won the prize";
+            feedback = "Correct! You won the prize.";
             emit Guessed(msg.sender, _number, true, feedback);
             _resetSecretNumber();
-            totalGuesses = 0; // Reset the total number of guesses made.
+            totalGuesses = 0;
         } else {
-        uint256 currentDifference = _number > secretNumber ? _number - secretNumber : secretNumber - _number;
+            uint256 currentDifference = _number > secretNumber ? _number - secretNumber : secretNumber - _number;
             if (lastResetTime > lastGuess[msg.sender]) {
                 if (currentDifference < lastDifference[msg.sender]) {
-                    feedback = "Your guess is warmer";
+                    feedback = "Warmer.";
```

Comment:
Is there a reason for changing the feedback message, @0xjba?

Reply:
Oh no not needed missed that, thanks for pointing will fix that